test/compile-error-test/CMakeLists.txt

```diff
@@ -1,77 +1,128 @@
 # Test if compile errors are produced where necessary.
 
 cmake_minimum_required(VERSION 3.1...3.18)
+project(compile-error-test)
 
-include(CheckCXXSourceCompiles)
-include(CheckCXXCompilerFlag)
+set(test_names "")
 
-set(CMAKE_REQUIRED_INCLUDES ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-set(CMAKE_REQUIRED_FLAGS ${CXX_STANDARD_FLAG} ${PEDANTIC_COMPILE_FLAGS})
-
-function (generate_source result fragment)
-  set(${result} "
-  #define FMT_HEADER_ONLY 1
-  #include \"fmt/format.h\"
-  int main() {
-    ${fragment}
-  }
-  " PARENT_SCOPE)
+function (expect_compile name code_fragment)
```

Comment:
Please add a comment documenting output variables, `test_names` and `tests_${name}_error_expected`.

Reply:
I added some comments, hope they are descriptive enough.